### PR TITLE
Update docs to use/download smaller sized artifacts for cli installation which excludes plugin binaries

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- This PR updates the installation documents to use small tarball/zip
  the file containing just the CLI to install the Tanzu CLI
- Also adds location to download admin-plugins from release artifacts

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: docs/cli/getting-started.md [hunk 1]
```diff
@@ -18,40 +18,42 @@ Following are the combinations supported for CLI
 
 #### Recommended method to install plugins (with API-driven plugin discovery activated)
 
-API driven plugin discovery feature is available with [v0.11.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.11.0) release as default method to install the plugins. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
+With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.14.0) release, more artifacts are getting published to allow users to download only the artifacts that are required for their needs.
```
Comment (navidshaikh) [inline docs/cli/getting-started.md:21]:
```suggestion
With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.14.0) release, the artifacts are better categorised and published to allow users to download and configure per recommended and alternate methods.
```

File: docs/cli/getting-started.md [hunk 2]
```diff
@@ -18,40 +18,42 @@ Following are the combinations supported for CLI
...
+
+The API driven plugin discovery feature is enabled as default method to install the plugins. So, most of the users will just need to install binary for Tanzu CLI. The plugins can be downloaded with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
```
Comment (navidshaikh) [inline docs/cli/getting-started.md:23]:
```suggestion
The API driven plugin discovery feature is enabled as default method to install the plugins. So, most of the users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
```

File: docs/cli/getting-started.md [hunk 3]
```diff
@@ -18,40 +18,42 @@ Following are the combinations supported for CLI
...
 
 ##### macOS/Linux
 
-- Download the latest [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest)
+- Download the latest tanzu-cli from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest). (`tanzu-cli-darwin-amd64.tar.gz`,  `tanzu-cli-linux-amd64.tar.gz`)
```
Comment (navidshaikh) [inline docs/cli/getting-started.md:27]:
```suggestion
- Download the latest tanzu-cli tarball from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest) page (`tanzu-cli-darwin-amd64.tar.gz` or `tanzu-cli-linux-amd64.tar.gz`)
```

File: docs/cli/getting-started.md [hunk 4]
```diff
@@ -68,12 +70,12 @@ API driven plugin discovery feature is available with [v0.11.0](https://github.c
 
 ##### Windows
 
-- Download the latest [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest)
+- Download the latest tanzu-cli from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest). (`tanzu-cli-windows-amd64.zip`)
```
Comment (navidshaikh) [inline docs/cli/getting-started.md:73]:
```suggestion
- Download the latest `tanzu-cli-windows-amd64.zip` from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest) page
```

File: docs/cli/plugin_implementation_guide.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ To add the builder plugin use `tanzu plugin install builder`
 
 With the [context-aware plugin discovery](/docs/design/context-aware-plugin-discovery-design.md) enabled, Tanzu CLI admin plugins should now be installed from local source as follows:
 
-1. Download the admin plugin tarball or zip file from the release artifacts and extract it
+1. Download the latest admin plugin tarball or zip file from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest). (`tanzu-framework-plugins-admin-linux-amd64.tar.gz`, `tanzu-framework-plugins-admin-darwin-amd64.tar.gz`, `tanzu-framework-plugins-admin-darwin-amd64.tar.gz`) and extract it. (Using `linux` as example OS for next steps)
```
Comment (navidshaikh) [inline docs/cli/plugin_implementation_guide.md:56]:
```suggestion
1. Download the latest admin plugin tarball or zip file from [release](https://github.com/vmware-tanzu/tanzu-framework/releases/latest) page (`tanzu-framework-plugins-admin-linux-amd64.tar.gz` or `tanzu-framework-plugins-admin-darwin-amd64.tar.gz` or `tanzu-framework-plugins-admin-darwin-amd64.tar.gz`) and extract it (using `linux` as example OS for next steps)
```

**Review by navidshaikh — CHANGES_REQUESTED** [review]
Since there are multiple artifacts on the release page, I think it'd be helpful to add a table in getting-started document describing the filename and one liner description for each artifact.

**Review by navidshaikh — APPROVED** [review]
lgtm!

As mentioned earlier, it'd be helpful to enlist all the artifacts we produce and one liner description of each (can be done in a follow up!).

File: docs/cli/getting-started.md [hunk 6]
```diff
@@ -18,40 +18,42 @@ Following are the combinations supported for CLI
 
 #### Recommended method to install plugins (with API-driven plugin discovery activated)
 
-API driven plugin discovery feature is available with [v0.11.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.11.0) release as default method to install the plugins. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
+With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.14.0) release, the artifacts are better categorised and published to allow users to download and configure per recommended and alternate methods.
```
Comment (vuil) [inline docs/cli/getting-started.md:21]:
```suggestion
With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.14.0) release, the artifacts are better categorized and published to allow users to download and configure per recommended and alternate methods.
```
```suggestion
With [v0.14.0](https://github.com/vmware-tanzu/tanzu-framework/releases/tag/v0.14.0) release, the artifacts are better categorised and published to allow users to download and configure per recommended and alternate methods.
```

File: docs/cli/getting-started.md [hunk 7]
```diff
@@ -18,40 +18,42 @@ Following are the combinations supported for CLI
...
+
+The API driven plugin discovery feature is enabled as default method to install the plugins. So, most of the users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
```
Comment (vuil) [inline docs/cli/getting-started.md:23]:
```suggestion
The API driven plugin discovery feature is enabled as default method to install the plugins. So, most users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
```
```suggestion
The API driven plugin discovery feature is enabled as default method to install the plugins. So, most of the users will just need to install the Tanzu CLI binary. The plugins can be installed and configured with `tanzu plugin sync` command. Learn more about this feature in the [design docs](../design/context-aware-plugin-discovery-design.md).
```

**Review by vuil — APPROVED** [review]
couple of language nits, should you choose to rebase, but otherwise lgtm.

**Review by vuil — APPROVED** [review]

**Comment by navidshaikh** [comment]
Logged the issue https://github.com/vmware-tanzu/tanzu-framework/issues/1459 for follow up. Merging this PR.

**Pull request merged**